watertap/unit_models/pressure_changer.py

```diff
@@ -11,35 +11,301 @@
+            #             (
+            #                     b.control_volume.properties_out[t].pressure
+            #                     / b.control_volume.properties_out[t].dens_mass_phase["Liq"]
+            #             )
+            #             - (
+            #                     b.control_volume.properties_in[t].pressure
+            #                     / b.control_volume.properties_in[t].dens_mass_phase["Liq"]
+            #             )
+            #     )
+        if self.config.variable_efficiency is not VariableEfficiency.none:
+            # replace the constant efficiency assumption using eta_ratio
+            # must be done after the eta_ratio expression is created.
+            @self.Constraint(self.flowsheet().time, doc="Actual pump efficiency")
+            def eta_constraint(b, t):
+                return b.efficiency_pump[t] == (b.bep_eta * b.eta_ratio[t])
+
+    def calculate_scaling_factors(self):
+        super().calculate_scaling_factors()
+
+        for ind, c in self.control_volume.isothermal_balance.items():
+            sf = iscale.get_scaling_factor(
+                self.control_volume.properties_in[0].temperature
+            )
+            iscale.constraint_scaling_transform(c, sf)
+
+        if hasattr(self, "bep_flow"):
+            if iscale.get_scaling_factor(self.bep_flow) is None:
+                iscale.set_scaling_factor(self.bep_flow, 10)
+
+        if hasattr(self, "bep_head"):
+            if iscale.get_scaling_factor(self.bep_head) is None:
+                iscale.set_scaling_factor(self.bep_head, 0.01)
+
+        if hasattr(self, "bep_eta"):
+            if iscale.get_scaling_factor(self.bep_eta) is None:
+                iscale.set_scaling_factor(self.bep_eta, 1)
+
+        if hasattr(self, "flow_ratio"):
+            if iscale.get_scaling_factor(self.flow_ratio) is None:
+                iscale.set_scaling_factor(self.flow_ratio, 1)
+
+        if hasattr(self, "head_ratio"):
+            if iscale.get_scaling_factor(self.head_ratio) is None:
+                iscale.set_scaling_factor(self.head_ratio, 1)
+
+        if hasattr(self, "efficiency_pump"):
+            if iscale.get_scaling_factor(self.efficiency_pump[0]) is None:
+                iscale.set_scaling_factor(self.efficiency_pump[0], 1)
+
+        # scale constraints
+
+        if hasattr(self, "flow_ratio_constraint"):
+            if iscale.get_scaling_factor(self.flow_ratio_constraint[0]) is None:
+                iscale.set_scaling_factor(self.flow_ratio_constraint[0], 1)
+
+        if hasattr(self, "head_ratio_constraint"):
+            if iscale.get_scaling_factor(self.head_ratio_constraint[0]) is None:
+                iscale.set_scaling_factor(self.head_ratio_constraint[0], 1)
+
+        if hasattr(self, "eta_constraint"):
+            if iscale.get_scaling_factor(self.eta_constraint[0]) is None:
+                iscale.set_scaling_factor(self.eta_constraint[0], 1)
+
```

Comment:
These are constraints, so we typically use constraint_scaling_transform. Also, we might be able to get away with omitting constraint scaling for these.

Reply:
Turns out it works well without constraint scaling.